routes/directories/Gamers.ts

```diff
@@ -0,0 +1,184 @@
+import {handleRecaptchaVerify} from '../recaptchaMiddleware';
+
+//===---==--=-=--==---===----===---==--=-=--==---===----//
+
+
+
+export async function createGamer(req: NextApiRequest, res: NextApiResponse) {
+    const newUser = req.body;
+    const recaptcha = await handleRecaptchaVerify(newUser.token);
+    const recaptchaSuccess = recaptcha.success;
+    const {username, platform} = newUser;
+    let error = null;
+
+    if (!username) {
+        error = {userMessage: 'body.username (String) is required', status: 400};
+    }
+
+    if (!platform) {
+        error = {userMessage: 'body.platform (String) is required', status: 400};
+    }
+
+    if (!recaptchaSuccess) {
+        error = {userMessage: 'failed recaptcha verification', status: 400};
+    }
+
+    if (!error) {
+        const gamerObj = {
+            username: username,
+            platform: platform
+        };
+
+        const gamers = await queryGamers(gamerObj);
+        const gamerExists = gamers.length > 0;
+
+        const gamerUrl = ['gamer', gamerObj.platform, encodeURIComponent(gamerObj.username)].join('/');
+
+        if (gamerExists) {
+            handleResponse(req, res, {userMessage: 'gamer already exists!', url: gamerUrl, gamer: gamers[0]});
+        } else {
+            try {
+                const initializedGamer = await initializeGamer(gamerObj);
+                const matches = await initializeMatches(initializedGamer);
+                handleResponse(req, res, {
+                    userMessage: 'gamer successfully added!',
+                    url: gamerUrl,
+                    gamer: initializedGamer,
+                    matches: matches
+                });
+            } catch (e) {
+                handleError(req, res, {userMessage: 'gamer not found'}, 400);
+            }
+        }
+    } else {
+        handleError(req, res, {message: error.message}, error.status);
+    }
+}
+
+
+
+export async function findGamers(req: NextApiRequest, res: NextApiResponse) {
+    const viewName = 'player_stat_summary';
+    const queryParams = req.query;
+    const offset = req.query.offset || 0;
+    const limit = req.query.limit || 10;
+    delete queryParams.offset;
+    delete queryParams.limit;
+    const rawGamerList = await queryView(viewName, queryParams, {offset, limit});
+    const sanitizedGamers = rawGamerList.map(sanitizeGamer);
+
+    handleResponse(req, res, sanitizedGamers);
+}
+
+
+
+const TEAMMATE_FILTER_KEYS = ['username', 'platform', 'aliases'];
+
+async function updateGamerUponRequest(gamerData) {
+    let gamerPromise = Bluebird.resolve(gamerData);
+
+    if (!gamerData.needs_update) {
+        gamerData.needs_update = true;
+        gamerPromise = updateGamer({username: gamerData.username, platform: gamerData.platform}, gamerData);
+    }
+
+    return await gamerPromise;
+}
+
+
+export async function getGamerDetails(req: NextApiRequest & { params: { username: string, platform: string } }, res: NextApiResponse) {
```

Comment:
Simply moved all of the individual route endpoints to this endpoint module, but we could re-organize them however we'd like